src/SupplierManager/SupplierManager.API/Startup.cs

```diff
@@ -0,0 +1,64 @@
+using AutoMapper;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.OpenApi.Models;
+using SupplierManager.API.ControllerFactory;
+
+namespace SupplierManager.API
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // services.AddScoped(typeof(IGenericRepository<>), typeof(GenericRepository<>));
+
+            services.AddControllers();
+            var mvcBuilder = services.AddMvc();
+            mvcBuilder.AddMvcOptions(o => o.Conventions.Add(new GenericControllerNameConvention()));
+            mvcBuilder.ConfigureApplicationPartManager(c =>
+            {
+                c.FeatureProviders.Add(new GenericControllerFeatureProvider());
+            });
+            var mapperConfig = new MapperConfiguration(
+                mc => mc.AddProfile(new MapperProfile())
+            );
+            services.AddSingleton(mapperConfig.CreateMapper());
+            services.AddControllers();
+
+            services.AddSwaggerGen(c =>
+            {
+                c.SwaggerDoc("v1", new OpenApiInfo {Title = "SupplierManager.API", Version = "v1"});
+            });
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
```

Comment:
Please validate mappings here or create tests, you can use [AssertConfigurationIsValid](https://docs.automapper.org/en/stable/Configuration-validation.html) method.